### PR TITLE
Fix GraphQLTestClient to work with both variables and files upload 

### PR DESCRIPTION
**Pull request by estyxx**
## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #1572 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release fixes an [issue](https://github.com/strawberry-graphql/strawberry/issues/1572)in the `GraphQLTestClient` when using both varaibles and files together.


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to @esterbeltrami for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Review by BryceBeagle — APPROVED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
+
+This release fixes an issue in the `GraphQLTestClient` when using both varaibles and files together.
```
Comment (DoctorJohn) [inline RELEASE.md:3]:
```suggestion
This release fixes an issue in the `GraphQLTestClient` when using both variables and files together.
```

**Review by DoctorJohn — CHANGES_REQUESTED** [review]
lgtm. About a week ago I started working on a PR which makes sure mixing upload and built in scalars works in all integrations. I'll submit it shortly as a follow up to this PR :)

File: strawberry/test/client.py [hunk 2]
```diff
@@ -149,7 +149,10 @@ def _build_multipart_file_map(
             else:
                 map[key] = [f"variables.{reference}"]
 
-        return map
+        # remove the other variables that are not files, we can't remove them before
+        # because they can be part of a list of files or folder
+        map_without_vars = {k: v for k, v in map.items() if k in files.keys()}
+        return map_without_vars
```
Comment (patrick91) [inline strawberry/test/client.py:155]: can you explain why we remove them in the comment? 😊

Comment (estyxx) [reply strawberry/test/client.py]: sure!

**Review by DoctorJohn — APPROVED** [review]

**Review by patrick91 — APPROVED** [review]

**Pull request merged**